### DISCUSSION TITLE
Unique filenames for singletons

### DISCUSSION THREAD

**Discussion by jaimeMF** [beetbox/beets#4436] [Q&A] [answered]
I know that for albums you can use %aunique for generating unique filenames for different albums. Is there something equivalent for singletons? I have a couple of standalone tracks from the same artist and with the same title, like for example [1](https://musicbrainz.org/recording/6dd4cdf6-15ff-4bc6-a1af-b7de4e9212e7) and [2](https://musicbrainz.org/recording/b68e1719-576c-43e8-b40f-e7a5ad761ef6),  and `beet move` always tries to rename them:
```sh
> beet move -p ugo
Moving 1 item (1 already in place).
Source                                                                                                  Destination
<LIBRARY>/Non-Album/Ugo Sforza/Fantasy on Veni Creator Spiritus, Op 7.1.mp3  -> <LIBRARY>/Non-Album/Ugo Sforza/Fantasy on Veni Creator Spiritus, Op 7.mp3
> beet move ugo
Moving 1 item (1 already in place).
> beet move -p ugo
Moving 1 item (1 already in place).
Source                                                                                                  Destination
<LIBRARY>/Non-Album/Ugo Sforza/Fantasy on Veni Creator Spiritus, Op 7.2.mp3  -> <LIBRARY>/Non-Album/Ugo Sforza/Fantasy on Veni Creator Spiritus, Op 7.mp3
> beet move ugo
Moving 1 item (1 already in place).
> beet move -p ugo
Moving 1 item (1 already in place).
Source                                                                                                  Destination
<LIBRARY>/Non-Album/Ugo Sforza/Fantasy on Veni Creator Spiritus, Op 7.1.mp3  -> <LIBRARY>/Non-Album/Ugo Sforza/Fantasy on Veni Creator Spiritus, Op 7.mp3
```

**Comment by sampsyo** [comment, marked as answer]
Very interesting! We don't currently have an analog to `%aunique`, which is a surprisingly complicated set of functionality, for singletons instead of albums. It would certainly be interesting to consider!

Barring that, one option would be to think of some distinguishing field you'd like to use in your filenames for these tracks. Then you could manually mark them as "ambiguous," using something like `beet modify ambig=1 Fantasy on Veni Creator Spiritus, Op 7`. Then, in the `paths:` section of your config, you could add an extra path template just for this kind of track:


```yaml
paths:
    ambig:1: Non-Album/$artist/$title - $catalognum
```

…or whatever, replacing `$catalognum` with the disambiguating field you want to use.

**Reply by jaimeMF** [reply]
Thanks, that's an option.

Since I would prefer to have something that works automatically I will try to implement a `%sunique` function.

**Reply by RollingStar** [reply]
Would it make more sense to have aunique work on arbitrary inputs (albums and singletons)?

**Reply by sampsyo** [reply]
#4438 ultimately shared a lot of code with `%aunique`, but it remains a separate path function.
